library/src/main/java/com/feedhenry/sdk/android/FHSyncService.java

```diff
@@ -0,0 +1,452 @@
+            for (Iterator<String> it = managedDatasetsJSON.keys(); it.hasNext(); ) {
+                String key = it.next();
+                JSONObject managedDatasetJSON = managedDatasetsJSON.getJSONObject(key);
+                JSONObject datasetSyncConfig = managedDatasetJSON.optJSONObject(KEY_DATASET_SYNC_CONFIG);
+                FHSyncConfig fhSyncConfig = null;
+                if (datasetSyncConfig != null) {
+                    fhSyncConfig = (new FHSyncConfig.Builder()).fromJSON(datasetSyncConfig).build();
+                }
+                JSONObject queryParams = managedDatasetJSON.optJSONObject(KEY_DATASET_QUERY_PARAMS);
+                JSONObject metadata = managedDatasetJSON.optJSONObject(KEY_DATASET_METADATA);
+                ManagedDatasetConfig datasetConfig = new ManagedDatasetConfig(fhSyncConfig, queryParams, metadata);
+                managedDatasets.put(key, datasetConfig);
+                log.d(GLOBAL_CONFIG, "Loaded config for dataset: " + key);
+            }
+        } catch (JSONException e) {
+            log.e(GLOBAL_CONFIG, "Error parsing sync config, using defaults.", e);
+            syncConfig = new FHSyncConfig.Builder().build();
+        }
+    }
+
+    private void saveSyncConfig() throws IOException {
+        try {
+            JSONObject configJSON = new JSONObject();
+            configJSON.put(KEY_SYNC_CONFIG, syncConfig.toJSON());
+            JSONObject managedDatasetsJSON = new JSONObject();
+            for (Map.Entry<String, ManagedDatasetConfig> entry : managedDatasets.entrySet()) {
+                ManagedDatasetConfig datasetConfig = entry.getValue();
+                JSONObject managedDatasetJSON = new JSONObject();
+                notNullDo(datasetConfig.syncConfig, config -> {
+                    try {
+                        managedDatasetJSON.put(KEY_DATASET_SYNC_CONFIG, config.toJSON());
+                    } catch (JSONException e) {
+                        throw new RuntimeException(e);
+                    }
+                });
+                managedDatasetJSON.put(KEY_DATASET_QUERY_PARAMS, datasetConfig.queryParams);
+                managedDatasetJSON.put(KEY_DATASET_METADATA, datasetConfig.metadata);
+                managedDatasetsJSON.put(entry.getKey(), managedDatasetJSON);
+            }
+            configJSON.put(KEY_MANAGED_DATASETS, managedDatasetsJSON);
+            configJSON.put(KEY_CLOUD_URL, cloudURL);
+            storage.putContent(GLOBAL_CONFIG, configJSON.toString().getBytes(UTF_8));
+            log.d(SERVICE_NAME, "Saved sync config persistently.");
+        } catch (JSONException e) {
+            log.e(SERVICE_NAME, "Unable to create sync config JSON.", e);
+        }
+    }
+
+    @Nullable
+    @Override
+    public IBinder onBind(Intent intent) {
+        log.d(SERVICE_NAME, "Service " + toString() + " bound.");
+        return binder;
+    }
+
+    @Override
+    protected void onHandleIntent(@Nullable Intent intent) {
+
```

Comment:
TBD: Add possibility to call by intent and using startService(). Will it make more sense for JobScheduler? Be careful with multithreading.